distribution/kernel/carbon-home/repository/resources/conf/templates/repository/conf/registry.xml.j2

```diff
@@ -28,8 +28,13 @@
     <registryRoot>/</registryRoot>
 
     <dbConfig name="wso2registry">
+    {% if database.carbon_db.enabled is sameas true %}
```

Comment:
I think this configuration should be other way. 

if nothing is set, then we should keep the existing behaviour (In any case, the default json config is missing, the behaviour should not change). So shall we change it?

same for the other configuration as well.

Reply:
Normlay, 
`{% if <config> is defined%} `
used to check a config existence and add/modify existing/default behavior.
Here we can't use that cause if the logic as follows
```
{% if database.carbon is defined%}
enabled carbon db 
```
(default) then we need to define database.carbon by default
for the other config, **if the config is available we need to remove something**, here also the logic fails due to the above reason
can u put a sample logic if possible

Reply:
Default.json is a mandatory file for configuration which by default is true.

Reply:
I updated the code without using the default json. https://github.com/wso2/carbon-kernel/pull/3599/commits/1165726fbfa0af1e8ca7cc3a0aedcaf4936c1ed3